docs/advanced-topics/tools-vs-workflows.rst

```diff
@@ -0,0 +1,117 @@
+| Nextflow               | N/A                                          | Any valid Nextflow workflow                         |
++------------------------+----------------------------------------------+-----------------------------------------------------+
+| Galaxy                 | N/A*                                         | Any valid Galaxy workflow                           |
++------------------------+----------------------------------------------+-----------------------------------------------------+
+
+\* There are tools that make up Galaxy workflows from the Galaxy toolbox or ToolShed.
+Dockstore currently does not support registration of these tools.
+
+
++------------------------+------------------------------------------+-------------------------------------------------+
+| Dockstore Support      | Tool                                     | Workflow                                        |
++========================+==========================================+=================================================+
+| "Launch with" support  |  Only via Dockstore CLI                  |  Dockstore CLI & "launch with" buttons on entry |
++------------------------+------------------------------------------+-------------------------------------------------+
+| Versioning             |  Based off of image's tags               |  Based off of branches/tags from Git repository,|
+|                        |                                          |                                                 |
+|                        |                                          |  even if one of the tasks executes in a Docker  |
+|                        |                                          |  image.                                         |
++------------------------+------------------------------------------+-------------------------------------------------+
+
+What about workflows that run in Docker containers?
+---------------------------------------------------
+
+Workflow descriptor files do not require a Docker container. That being said, you can still specify external Docker images
+within the descriptor files, and doing so is generally recommended. Each task (WDL) or step (CWL) can specify a Docker image, which might be totally different from the Docker image used by a previous task/step in the workflow. Because of this, workflows registered on Dockstore that reference Docker image(s) will still follow versioning from GitHub, instead of versioning based on the tags of the Docker image(s).
+
+.. hint::
+    The Cromwell execution engine requires that WDL task specify a Docker image when running on GCP or AWS backends, including Terra. Therefore, if your WDL workflow needs to run on Terra, make sure that each task uses the Docker runtime attribute. For more information see the `Cromwell docs <https://cromwell.readthedocs.io/en/stable/RuntimeAttributes/#docker>`_.
```

Comment:
> WDL task specify a Docker image when running on GCP or AWS backends

Do you mean if you're running Cromwell without Terra? Sounds like... I'm not sure in all cases Cromwell would know it's running on AWS (you're just on EC2 instance), but probably not worth getting into that.

Reply:
Strictly speaking, [Cromwell's docs say this about the docker runtime attribute](https://cromwell.readthedocs.io/en/stable/RuntimeAttributes/#docker):

> GCP: This attribute is mandatory when submitting tasks to Google Cloud.
> AWS Batch: This attribute is mandatory when submitting tasks to AWS Batch.

Due to how the docs tend to present things as "this is what changes based on the backend," I interpreted this to mean "any AWS backend," but I will be the first to admit I know very little about AWS. What's your interpretation?

Reply:
My interpretation is that if you use Cromwell with AWS Batch, which will let Cromwell scale up on its own, then you need it.

But if you just ssh into a machine that happens to be hosted on AWS and running Cromwell locally, it doesn't matter.

Same is probably true for GCP...

So, I would either drop the the Cromwell execution engine reference, or if you want to refer to it, maybe talk about in terms of [scaling](https://cromwell.readthedocs.io/en/stable/Scaling/).

Reply:
Good point -- I clarified it.